### DISCUSSION TITLE
OnePlus 11 OxygenOS 14

### DISCUSSION THREAD

**Discussion by alecchambers94** [tiann/KernelSU#1209] [General]
I see a lot of people are having issues with OxygenOS 14. I flashed the December version Android-13.5.15.137 and it worked for me.
![Screenshot_2023-12-13-10-19-50-03_baaa51257c82a8b3f281ce09e6f2493f](https://github.com/tiann/KernelSU/assets/19499380/df404038-78af-4e98-b2d6-78abe3134f13)

**Comment by tiann** [comment]
Do you clear the `/data` partition after flash?

**Reply by alecchambers94** [reply]
I did not need to but I did. I tried all of the newer kernels and even back to Android-13.5.15.94 on OxygenOS 14. Working perfect. The ones you download from the main page still bootloop and do not work at all for OOS 14
![Screenshot_2023-12-14-11-36-10-81_baaa51257c82a8b3f281ce09e6f2493f](https://github.com/tiann/KernelSU/assets/19499380/8941a291-5ce4-47b3-ada4-1b0c665410d1)

**Comment by ashoktvm** [comment]
@tiann Please release CI for Android12-9

**Reply by tiann** [reply]
No that branch yet.

**Reply by ashoktvm** [reply]
Means? it will not be released?

**Reply by tiann** [reply]
The GKI branch is not released, don't know why, ask Google.

**Reply by ashoktvm** [reply]
Oh i thought it was for kernelSU only. My bad

**Reply by ashoktvm** [reply]
Does this mean that until google releases next update the next version will not have support for Android 12-9

**Comment by ashoktvm** [comment]
Looks like android 12 has changed to bimonthly basis

```Android 12 GKI Release
After May 2023, the android12-5.10 GKI releases are on a 2-month cadence and published mid-month. The following table is applicable only to [android12-5.10](https://source.android.com/docs/core/architecture/kernel/gki-android12-5_10-release-builds).```

**Comment by alecchambers94** [comment]
If anyone still is unable to get KernelSU working on OxygenOS 14 here is the kernels I used, tested all multiple times. 
[android13-5.15.104_2023-07-boot.img.gz](https://github.com/tiann/KernelSU/files/13676076/android13-5.15.104_2023-07-boot.img.gz)
[android13-5.15.119_2023-09-boot.img.gz](https://github.com/tiann/KernelSU/files/13676077/android13-5.15.119_2023-09-boot.img.gz)
[android13-5.15.123_2023-11-boot.img.gz](https://github.com/tiann/KernelSU/files/13676078/android13-5.15.123_2023-11-boot.img.gz)
[android13-5.15.137_2023-12-boot.img.gz](https://github.com/tiann/KernelSU/files/13676080/android13-5.15.137_2023-12-boot.img.gz)
[android13-5.15.41_2022-11-boot.img.gz](https://github.com/tiann/KernelSU/files/13676082/android13-5.15.41_2022-11-boot.img.gz)
[android13-5.15.74_2023-01-boot.img.gz](https://github.com/tiann/KernelSU/files/13676083/android13-5.15.74_2023-01-boot.img.gz)
[android13-5.15.78_2023-03-boot.img.gz](https://github.com/tiann/KernelSU/files/13676084/android13-5.15.78_2023-03-boot.img.gz)
[android13-5.15.94_2023-05-boot.img.gz](https://github.com/tiann/KernelSU/files/13676088/android13-5.15.94_2023-05-boot.img.gz)

**Comment by ashoktvm** [comment]
> If anyone still is unable to get KernelSU working on OxygenOS 14 here is the kernels I used, tested all multiple times. [android13-5.15.104_2023-07-boot.img.gz](https://github.com/tiann/KernelSU/files/13676076/android13-5.15.104_2023-07-boot.img.gz) [android13-5.15.119_2023-09-boot.img.gz](https://github.com/tiann/KernelSU/files/13676077/android13-5.15.119_2023-09-boot.img.gz) [android13-5.15.123_2023-11-boot.img.gz](https://github.com/tiann/KernelSU/files/13676078/android13-5.15.123_2023-11-boot.img.gz) [android13-5.15.137_2023-12-boot.img.gz](https://github.com/tiann/KernelSU/files/13676080/android13-5.15.137_2023-12-boot.img.gz) [android13-5.15.41_2022-11-boot.img.gz](https://github.com/tiann/KernelSU/files/13676082/android13-5.15.41_2022-11-boot.img.gz) [android13-5.15.74_2023-01-boot.img.gz](https://github.com/tiann/KernelSU/files/13676083/android13-5.15.74_2023-01-boot.img.gz) [android13-5.15.78_2023-03-boot.img.gz](https://github.com/tiann/KernelSU/files/13676084/android13-5.15.78_2023-03-boot.img.gz) [android13-5.15.94_2023-05-boot.img.gz](https://github.com/tiann/KernelSU/files/13676088/android13-5.15.94_2023-05-boot.img.gz)

Which phone exactly?

**Reply by alecchambers94** [reply]
Oh sorry I'm using One Plus 11 CPH2451

**Reply by alecchambers94** [reply]
But with those kernels any phone that uses Android 13-5.15.XX should work properly. If it works on the higher security patch versions but has weird issues/glitches out and freezes or takes forever to open stuff you can just flash the next lower version until you find a stable one. It's usually the drivers that are hard coded in your phone from the manufacturer that decides on how high you can go.

**Reply by DragoWing** [reply]
I tried everything but it didn't work for me on 2449 A14, even wiped everything, A13 worked just fine, it keeps going to boot loop whatever I do, 😭😭😭😭

**Comment by ashoktvm** [comment]
> I tried everything but it didn't work for me on 2449 A14, even wiped everything, A13 worked just fine, it keeps going to boot loop whatever I do, 😭😭😭😭

same thing happened, even on A13, but with latest versions

**Reply by DragoWing** [reply]
Yeah, so how did people get it to work tho, I tried everything but it just won't boot, 😅

**Reply by ashoktvm** [reply]
No idea. Tried fastboot -w for formatting data and still not fixed. I am using 11R though with 5.10 kernel

**Reply by DragoWing** [reply]
Yeah, I tried everything too, I gave up today eventually,  I am 11 on 5.15 kernel

**Comment by Mario6425** [comment]
I have exactly the same problem.
I have gone through all the images with all the variations. 
kernelsu cph2449 oxygenos 14 NO CHANCE

**Reply by DragoWing** [reply]
Yeah, that's so sad it doesn't work, now we're stuck with magisk

**Reply by Mario6425** [reply]
I like kitsune mask, formerly magisk delta, what huskydg does there is bomb. I find Kernelsu too incompatible with many modules, it still needs time. But I was curious about the attempt to root the kernel via Apatch. Hence the search for the Boot.img

**Reply by DragoWing** [reply]
I am using kitsune now, but something happens cuz they deleted their telegram with a weird message, but if it needs the same conditions then how come apatch works but kernelsu won't, 😅

**Comment by ashoktvm** [comment]
> Yeah, that's so sad it doesn't work, now we're stuck with magisk

There is Apatch

**Reply by DragoWing** [reply]
Hmmm, I've heard it's not as good, is it? 😅

**Reply by Mario6425** [reply]
Yes, see my comment above. To work with Apatch, you need the same conditions as for kernelsu. A bit rash but thanks anyway :)

**Comment by ashoktvm** [comment]
> Hmmm, I've heard it's not as good, is it? 😅

I have tried it. And I won't say it is as good as KernelSU, but it's getting there. It's just been announced start of this month. But considering the time its being developed it's pretty good

**Reply by DragoWing** [reply]
Oh, cool, so I assume it works different then kernelsu assuming it works, so I guess it's not kernel based?

**Reply by ashoktvm** [reply]
No it works almost like kernelsu, but the benefit is you can patch boot img wih apatch to make it work. Like Magisk. But the benefit is it works on kernel from 3.18 to 6.1

**Reply by DragoWing** [reply]
Then I wonder how they can make it work but kernelsu can't, is it better then magisk? XD

**Reply by ashoktvm** [reply]
Cant compare it now with magisk since its initial stages

**Reply by DragoWing** [reply]
I am using it now, can't say I
I dislike it, XD
